src/runtime_src/core/edge/drm/zocl/zocl_rpu_channel.c

```diff
@@ -183,22 +185,45 @@ static int zchan_cmd_log_apu_log(struct zocl_rpu_channel *chan, u32 add_off, u32
 	u32 offset, u32 *total_countp)
 {
 	u32 count = 0;
+	struct file *fp;
+	loff_t fp_size = 0;
+	loff_t fp_offset = 0;
+	int ret = 0;
+	u32 remain_size = 0;
 
-	memset(info_buf, 0, sizeof(info_buf));
-	/*
-	 * Example code of collecting data based on request size and offset
-	 */
-	if (offset == 0) {
-		count = snprintf(info_buf, sizeof(info_buf), "first 4096");
-		memcpy_toio(chan->mem_base + add_off, info_buf, count);
-		count = 4096;
-	} else if (offset == 4096) {
-		count = snprintf(info_buf, sizeof(info_buf), "second 4096");
-		memcpy_toio(chan->mem_base + add_off, info_buf, count);
-		count = 4096;
+	fp = filp_open("/var/log/messages", O_RDONLY, 0);
+	if (IS_ERR(fp)) {
+		*total_countp = 0;
+		pr_err("Can't open /var/log/messages\n");
+		return -EINVAL;
+	}
+
+	fp_size = i_size_read(file_inode(fp));
+	fp_offset = offset;
+	remain_size = fp_size - offset;
+	// Send chunks of log
+	if (remain_size >= size) {
+		ret = kernel_read(fp, &log_buf, size, &fp_offset);
+		if (!ret) {
+			pr_err("Can't read from /var/log/messages\n");
+			*total_countp = 0;
+			return -EINVAL;
```

Comment:
I think we need to close the fp before return.

Reply:
updated